docs/DeveloperGuide.md

```diff
@@ -233,6 +233,12 @@ From the notes in the sequence diagram above, for every attribute in the instrum
 prompt to get user to provide information for that attribute. This is done through a series of methods in
 the `TextUi` class.
 
+### Edit instrument feature
+<img src="images/EditInstrumentSequenceDiagram.png" width="600"/>
+<img src="images/EditRefrence1.png" width="600"/>
+<img src="images/EditRefrence2.png" width="600"/>
```

Comment:
<img width="552" alt="ss8" src="https://user-images.githubusercontent.com/60163283/140519016-2ed80460-f69b-4a84-8417-a90fae98ae93.PNG">
Change to loop until valid ... is entered. The loop box is in the wrong place. The loop box should be in the editNameParameter method and editXYZParameter. And the editNameParameter should have an alt where the first condition is if the attribute is found in the hashset, else it then does the loop.
So the loop box should be within the second box below alt in the editNameParameter.
^same for editXYZParameter. Can also add a note stating what XYZ stands for like list down all the parameters currentPrice, sentiment and so on and replace your current note about the box. Its not clear what box is referring to and doesn't seem the right term to use.